src/services/faculty_service.py

```diff
@@ -0,0 +1,36 @@
+from src.repositories.faculty_repository import FacultyRepository
+from src.models.faculty import Faculty
+
+
+class FacultyService:
+    def __init__(self):
+        self.repo = FacultyRepository()
+
+    def get_all_faculties(self):
+        faculties = self.repo.get_all_faculties()
+        return [f.to_dict() for f in faculties]
+
+    def get_faculty_by_id(self, faculty_id):
+        faculty = self.repo.get_faculty_by_id(faculty_id)
+        return faculty.to_dict() if faculty else None
+
+    def create_faculty(self, faculty: Faculty):
+        created = self.repo.create_faculty(faculty)
+        return created.to_dict()
+
+    def update_faculty(self, faculty_id: int, updated_faculty: Faculty):
```

Comment:
you should only accept Faculty no need for id to accept as it is contained in Faculty